ci/build.yml

```diff
@@ -185,13 +224,26 @@ jobs:
     - da_ghc_lib
     - check_for_release
     - patch_bazel_windows
+    - git_sha
   variables:
-    release_sha: $[ dependencies.check_for_release.outputs['out.release_sha'] ]
-    release_tag: $[ coalesce(dependencies.check_for_release.outputs['out.release_tag'], '0.0.0') ]
-    trigger_sha: $[ dependencies.check_for_release.outputs['out.trigger_sha'] ]
-    is_release: $[ dependencies.check_for_release.outputs['out.is_release'] ]
-    skip_tests: $[ and(eq(variables.is_release, 'true'),
+    - name: release_sha
+      value: $[ dependencies.check_for_release.outputs['out.release_sha'] ]
+    - name: release_tag
+      value: $[ coalesce(dependencies.check_for_release.outputs['out.release_tag'], '0.0.0') ]
+    - name: trigger_sha
+      value: $[ dependencies.check_for_release.outputs['out.trigger_sha'] ]
+    - name: is_release
+      value: $[ dependencies.check_for_release.outputs['out.is_release'] ]
+    - name: skip_tests
+      value: $[ and(eq(variables.is_release, 'true'),
                        eq(variables['Build.SourceBranchName'], 'main')) ]
+    - name: git_branch_sha
+      value: $[ dependencies.git_sha.outputs['out.branch'] ]
+    - name: git_main_sha
+      value: $[ dependencies.git_sha.outputs['out.main'] ]
+    - name: git_fork_point_sha
+      value: $[ dependencies.git_sha.outputs['out.fork_point'] ]
+    - template: job-variables.yml
```

Comment:
How does this work? I would have thought this just gets inlined and via have
```
variables:
  - name: release_sha
    value: …
  - variables:
    - name: pipeline_timestamp
      value: …
```
Are nested variable blocks a thing?

Reply:
If taken that syntax from the docs [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#variable-reuse). As I understand it `template` has slightly different meanings in different contexts in that it's not just a string substitution.

Reply:
lovely thanks for the explanation

Reply:
> As I understand it `template` has slightly different meanings in different contexts in that it's not just a string substitution.

AFAICT it's always a string substitution; the surprising part about it is _when_ that substitution happens compared to other substitutions (`$()`, `$[[]]`, etc.).

Reply:
@garyverhaegen-da I meant that the contents of `job-variables.yml` are not just pasted in as in
```
variables:
  - name: release_sha
    value: …
  - variables:
    - name: pipeline_timestamp
      value: …
```

But, instead, it expects a `variables` map-item in the template file and inserts these variables as in 
```
variables:
  - name: release_sha
    value: …
  - name: pipeline_timestamp
    value: …
```

Reply:
Oh, right. Yes, there are different types of templates. Details [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#insert-a-template); generally speaking, if your template contains multiple jobs, you start it with `jobs:` and you can use it in `jobs:`; if it has multiple steps, you start it with `steps:` and you can use it in `steps:`, etc.